### PR TITLE
Adding vnet table to the pipeline

### PR DESCRIPTION
**Pull request by vijasrin**
- ENI table drives vnet-id for the VNET table as specified in the GNMI
schema
- Route lookup drives destination vnet-id for the VNET table lookup
- Removing redundant eni_to_vni table

### CODE REVIEW

File: dash-pipeline/tests/vnet_out/vnet_out.cpp [hunk 1]
```diff
@@ -68,25 +63,6 @@ int main(int argc, char **argv)
 
     attrs.clear();
 
-    sai_outbound_eni_to_vni_entry_t e2v = {};
```
Comment (marian-pritsak) [inline dash-pipeline/tests/vnet_out/vnet_out.cpp:71]: Substitute with creating a VNET object?

**Review by marian-pritsak — CHANGES_REQUESTED** [review]

File: dash-pipeline/bmv2/dash_metadata.p4 [hunk 2]
```diff
@@ -36,6 +36,8 @@ struct metadata_t {
     direction_t direction;
     encap_data_t encap_data;
     EthernetAddress eni_addr;
+    bit<16> vnet_id;
```
[the remark this answers is not in the document]
Comment (vijasrin) [reply dash-pipeline/bmv2/dash_metadata.p4:39]: following the existing eni_id which is also 16 bits. Should we address converting all IDs to 32 bit in a separate PR ?

**Comment by marian-pritsak** [comment]
Please resolve conflicts

**Comment by vijasrin** [comment]
@marian-pritsak hitting SAI lint error with this change:
`ERROR: unknown object type 'SAI_OBJECT_TYPE_VNET' on SAI_ENI_ATTR_VNET_ID
ERROR: unknown object type 'SAI_OBJECT_TYPE_VNET' on SAI_OUTBOUND_ROUTING_ENTRY_ATTR_VNET_ID`

The object type is generated as SAI_OBJECT_TYPE_OUTBOUND_VNET. We need the autogen changes from [https://github.com/Azure/DASH/pull/152/files](url) and add an exception for this table

**Review by prsunny — APPROVED** [review]
lgtm.

**Comment by marian-pritsak** [comment]
@vijasrin Please resolve conflicts

**Comment by vijasrin** [comment]
@marian-pritsak Fixed conflicts and added vnet table create/del to the tests

```

[14:59:50.290] [bmv2] [D] [thread 27] Entry 0 added to table 'dash_ingress.outbound.vnet|dash_vnet'
[14:59:50.290] [bmv2] [D] [thread 27] Dumping entry 0
Match key:
* meta.vnet_id:vnet_id: EXACT     0003
Action entry: dash_ingress.outbound.set_vnet_attrs - 90000,
```

File: dash-pipeline/SAI/sai_api_gen.py [hunk 3]
```diff
@@ -380,6 +380,8 @@ def write_sai_files(sai_api):
                 for table_name in all_table_names:
                     if table_ref.endswith(table_name):
                         param[OBJECT_NAME_TAG] = table_name
+                    if table_name.endswith(table_ref):
```
Comment (marian-pritsak) [inline dash-pipeline/SAI/sai_api_gen.py:383]: Why doesn't the upper condition work?

**Review by marian-pritsak — CHANGES_REQUESTED** [review]

File: dash-pipeline/SAI/sai_api_gen.py (hunk 3 above)
Comment (vijasrin) [reply dash-pipeline/SAI/sai_api_gen.py:383]: this is required to generate correct object id for the attribute reference. Without this fix:
```
`260     /**
261      * @brief Action route_vnet parameter VNET_ID
262      *
263      * @type sai_object_id_t
264      * @flags CREATE_AND_SET
265      * @objects SAI_OBJECT_TYPE_ <--------------
266      * @allownull true
267      * @default SAI_NULL_OBJECT_ID
268      * @validonly SAI_OUTBOUND_ROUTING_ENTRY_ATTR_ACTION == SAI_OUTBOUND_ROUTING_ENTRY_ACTION_ROUTE_VNET
269      */
270     SAI_OUTBOUND_ROUTING_ENTRY_ATTR_VNET_ID,```

Comment (marian-pritsak) [reply dash-pipeline/SAI/sai_api_gen.py:383]: My question is about upper condition. Why `if table_ref.endswith(table_name)` does not catch that? What is `table_ref` and `table_name` equal to in your case?

Comment (vijasrin) [reply dash-pipeline/SAI/sai_api_gen.py]: @marian-pritsak fixed to use the table name for only the user portion as discussed offline. Dont need this change now.

Comment (vijasrin) [reply dash-pipeline/SAI/sai_api_gen.py]: @marian-pritsak added the fix for the key names as well. Plz check

**Review by marian-pritsak — APPROVED** [review]

**Comment by marian-pritsak** [comment]
Pending API approval https://github.com/opencomputeproject/SAI/pull/1562

**Pull request merged**